Add a warning when Ads' currency is different

tomalec opened the pull request:
### Changes proposed in this Pull Request:
Add a warning when Ads' currency is different than the store's one.

Addresses https://github.com/woocommerce/google-listings-and-ads/issues/363#issuecomment-876309170.


### Screenshots:

<!--- Optional --->
![warning](https://user-images.githubusercontent.com/17435/132068374-26823d52-a516-4ec8-ac8b-d64f9b245cd9.gif)



### Detailed test instructions:

1. Setup Ads account with a currency - _X_ (PLN)
3. Change the store currency to _Y_ (EUR)
4. Go to the 
	- Dashboard ([`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard`](https://gla1.test/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fdashboard) )
	- Program reports ([`/wp-admin/admin.php?page=wc-admin&reportKey=programs&path=%2Fgoogle%2Freports`](https://gla1.test/wp-admin/admin.php?page=wc-admin&reportKey=programs&path=%2Fgoogle%2Freports))
	- Product report ([`/wp-admin/admin.php?page=wc-admin&reportKey=products&path=%2Fgoogle%2Freports`](https://gla1.test/wp-admin/admin.php?page=wc-admin&reportKey=products&path=%2Fgoogle%2Freports)
5. Check the warning that will eventually appear.


<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->
### Changelog entry

> Add - Warning notice when the Ads' currency is different from the store's one.

### Additional notes:
- There is no external link icon at "Read more" as I'd add it in a separate PR for all the external/documentation links. See https://github.com/woocommerce/google-listings-and-ads/issues/984, woocommerce/woocommerce#32236

tomalec committed:
Add a warning when Ads' currency is different
than the store's one.

Addresses https://github.com/woocommerce/google-listings-and-ads/issues/363#issuecomment-876309170.

eason9487 commented on js/src/components/different-currency-notice.js:
```diff
@@ -0,0 +1,59 @@
+							className="gla-different-currency-notice__link"
+							href="https://support.google.com/google-ads/answer/9841530"
+							eventName="gla_different_currency_notice_link_click"
+							context="dashboard"
```
❓ 

This notice is used on different pages, so the `context` value would probably be changed along with pages.

tomalec replied:
Addressed in https://github.com/woocommerce/google-listings-and-ads/pull/994/commits/13d77c233db82d5bcc3ce88dbfec6a27004b29ab

tomalec committed:
Add `context` property for `DifferentCurrencyNotice`,

to forward a different `context` event prop from different pages.

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/994#discussion_r702530270.

tomalec commented:
I addressed the comment, and the PR is ready for another round.

eason9487 reviewed (approved):
Tested locally and works well. LGTM.
